soepy/python/solve/solve_auxiliary.py

```diff
@@ -206,31 +212,39 @@ def pyth_backward_induction(attr_dict, state_space_args):
     return periods_emax
 
 
-def construct_covariates(attr_dict, states_all, period, k):
+def construct_covariates(
+    states_all, states_number_period, max_states_period, attr_dict
+):
     """Constructs additional covariates given state space components."""
 
-    # Determine education level given number of years of education
-
-    # Unpack attributes from the model specification
+    # Unpack attributes from the model specification:
+    num_periods = attr_dict["GENERAL"]["num_periods"]
     educ_min = attr_dict["INITIAL_CONDITIONS"]["educ_min"]
 
-    # Unpack state space components
-    educ_years = states_all[period, k, 0]
+    # Initialize covariates array
+    covariates = np.tile(MISSING_INT, (num_periods, max_states_period, 4))
 
-    # Extract education information
-    if educ_years <= 10:
-        educ_level = [1, 0, 0]
+    # Fill in education information
+    for period in range(num_periods):
 
-    elif (educ_years > 10) and (educ_years <= 12):
-        educ_level = [0, 1, 0]
+        for k in range(states_number_period[period]):
 
-    else:
-        educ_level = [0, 0, 1]
+            educ_years = states_all[period, k, 0]
 
-    educ_years_idx = educ_years - educ_min
+            # Extract education information
+            if educ_years <= 10:
```

Comment:
Please create local variables that increase the readability of the if statements. For example, we could create `is_middle_school = (educ_years > 10) and (educ_years <= 12)` or something to that effect and then use `is_middle_school` later. Also, please only use `else:` to throw an error. We want to deal with all if - possibilities explicitly through 'elseif`s. All this will increase the readability considerably as we add more and more covariates to our model.

Reply:
I would prefer to wait with the creation of variables such as is_middle_school until I see the data and talk to Peter about how to handle education in particular and all other covariates in the model in general. Then I can pick appropriate names and might have to change the code a bit anyhow.

Reply:
Fine, but please ensure already that else raises an error.